### PR TITLE
CORDA-2586 explorer exception handling

### PR DESCRIPTION
**Pull request by StefanIliev545**
Changed the dialog used for exception handling in the login view from the default ExceptionDialog which always shows a stack trace to a custom Dialog that won't show stack trace unless asked to. RPC exceptions won't show a stack trace as from my understanding running through the code that throws them, they are either network or protocol issues, thus there isn't much point. Other exceptions that reach the catch will still be shown. The Dialog should display a hyperlink to whatever is generated by errorCodeUrl() extension function. (Only tested that it works on mac) 

CordaVersionProvider now contains a CordaVersion, so that the error url generation logic could be moved to a different library, because CordaVersionProvider extends some  third party module interface and there is no point in linking that interface whenever you want to get the versioning info.

Note that I'm still navigating around gradle and the build.gradle in common-logging was copied from another common/ lib so there might be an obvious mistake there or something useless.

### CODE REVIEW

File: common/logging/src/main/kotlin/net/corda/common/logging/CordaVersion.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package net.corda.common.logging
+
+import com.jcabi.manifests.Manifests
+
+/**
+ * Simple version printing when command is called with --version or -V flag. Assuming that we reuse Corda-Release-Version and Corda-Revision
+ * in the manifest file.
+ */
+class CordaVersion  {
```
Comment (josecoll) [inline common/logging/src/main/kotlin/net/corda/common/logging/CordaVersion.kt]: This information already exists elsewhere in the code base.
Please do not duplicate.

Comment (StefanIliev545) [reply common/logging/src/main/kotlin/net/corda/common/logging/CordaVersion.kt]: Thanks, I've removed it in the last commit.

File: common/logging/src/main/kotlin/net/corda/common/logging/ExceptionsErrorCodeFunctions.kt [hunk 2]
```diff
@@ -1,21 +1,22 @@
-package net.corda.cliutils
+package net.corda.common.logging
 
 import org.apache.logging.log4j.Level
 import org.apache.logging.log4j.message.Message
 import org.apache.logging.log4j.message.SimpleMessage
 import java.util.*
 
-internal fun Message.withErrorCodeFor(error: Throwable?, level: Level): Message {
+
```
Comment (anthonykeenan) [inline common/logging/src/main/kotlin/net/corda/common/logging/ExceptionsErrorCodeFunctions.kt]: extra line

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt [hunk 3]
```diff
@@ -6,6 +6,8 @@ import org.apache.logging.log4j.core.appender.rewrite.RewritePolicy
 import org.apache.logging.log4j.core.config.plugins.Plugin
 import org.apache.logging.log4j.core.config.plugins.PluginFactory
 import org.apache.logging.log4j.core.impl.Log4jLogEvent
+import net.corda.common.logging.withErrorCodeFor
+
```
Comment (anthonykeenan) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt]: extra line

File: tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt [hunk 4]
```diff
@@ -6,6 +6,8 @@ import org.apache.logging.log4j.core.appender.rewrite.RewritePolicy
...
 
 @Plugin(name = "ErrorCodeRewritePolicy", category = Core.CATEGORY_NAME, elementType = "rewritePolicy", printObject = false)
 class ErrorCodeRewritePolicy : RewritePolicy {
```
Comment (anthonykeenan) [inline tools/cliutils/src/main/kotlin/net/corda/cliutils/ErrorCodeRewritePolicy.kt]: Shouldn't this go in the logging module?

**Pull request merged**